### PR TITLE
feat(notifier): fix provider validation

### PR DESCRIPTION
**Pull request by vjgene**
Fix bug when adding a fresh new provider. hasPlans() returns false always as checking Boolean(plans.length) on json object fails.

This fix uses two step validation, one is for correct url and other is for validating of actual SubscriptionPlanDTO content length in the response. It also separates NotifierResponse and SubscriptionPlanResponse as NotifierResponse can be extended by other Notifier response entities.

### CODE REVIEW

File: src/api/rif-notifier-service/subscriptionPlans.ts [hunk 1]
```diff
@@ -8,18 +9,19 @@ export default class SubscriptionPlans
...
-    const plans = await this._fetch()
-      .catch(() => {
-        this.errorReporter({
-          error: new Error('No available subscription plan'),
-          text: 'Atleast one subscription plan must be available for registration',
-          id: 'service-fetch',
-        })
-        return []
+  hasPlans = ({ content: { length }, message }: SubscriptionResponse): boolean => {
+    const validResponse = message === NOTIFIER_RESPONSE_MESSAGES.OK
```
Comment (jurajpiar) [inline src/api/rif-notifier-service/subscriptionPlans.ts]:
```
    const isValidResponse = message === NOTIFIER_RESPONSE_MESSAGES.OK
```

File: src/components/organisms/notifier/ProviderRegistrar.tsx [hunk 2]
```diff
@@ -27,11 +29,13 @@ const ProviderRegistrar: FC<ProviderRegistrarProps> = ({
...
 
-  const validateProviderURL = (url: string): Promise<boolean> => {
+  const validateProviderURL = async (url: string): Promise<boolean> => {
     const notifierService: SubscriptionPlans = new SubscriptionPlans(url)
     notifierService.connect(reportError)
 
-    return notifierService.hasPlans()
+    const response: SubscriptionResponse = await notifierService.fetch()
+    const { message } = response
+    return message === NOTIFIER_RESPONSE_MESSAGES.OK && notifierService.hasPlans(response)
```
Comment (jurajpiar) [inline src/components/organisms/notifier/ProviderRegistrar.tsx]: seem to be doing the same job twice. why not fetching inside the hasPlans as we did before?

File: src/api/rif-notifier-service/subscriptionPlans.ts (hunk 1 above)
Comment (jurajpiar) [inline src/api/rif-notifier-service/subscriptionPlans.ts]:
```suggestion
  hasPlans = (): boolean => {
   const { content: { length }, message }: SubscriptionResponse = await this.fetch()
    const isValidResponse = message === NOTIFIER_RESPONSE_MESSAGES.OK
```

File: src/api/rif-notifier-service/interfaces.ts [hunk 3]
```diff
@@ -9,4 +14,17 @@ export abstract class NotifierAPIService extends AbstractAPIService {
...
+      .catch((error) => {
+        this.errorReporter({
+          error,
+          text: 'Error fetching data from notifier',
+          id: 'service-fetch',
+        })
+        return { content: [], message: NOTIFIER_RESPONSE_MESSAGES.ERROR }
+      })
+    return response
+  }
```
Comment (jurajpiar) [inline src/api/rif-notifier-service/interfaces.ts]:
```suggestion
  fetch = (): Promise<NotifierResponse> => this._fetch()
      .catch((error) => {
        this.errorReporter({
          error,
          text: 'Error fetching data from notifier',
          id: 'service-fetch',
        })
        return { content: [], message: NOTIFIER_RESPONSE_MESSAGES.ERROR }
      })
```

File: src/api/rif-notifier-service/responseConstants.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const NOTIFIER_RESPONSE_MESSAGES = {
+  OK: 'OK',
+  ERROR: 'ERROR',
+} as const
+
+export type NotifierResponseMessage = keyof typeof NOTIFIER_RESPONSE_MESSAGES
```
Comment (jurajpiar) [inline src/api/rif-notifier-service/responseConstants.ts]: move to interfaces

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Thanks for fixing the issue. Got some requests tho.

File: src/components/organisms/notifier/ProviderRegistrar.tsx (hunk 2 above)
Comment (vjgene) [reply src/components/organisms/notifier/ProviderRegistrar.tsx]: moved

**Review by jurajpiar — APPROVED** [review]

File: src/api/rif-notifier-service/subscriptionPlans.ts [hunk 5]
```diff
@@ -8,18 +9,20 @@ export default class SubscriptionPlans
...
-          id: 'service-fetch',
-        })
-        return []
+    const { message, content: { length } }: SubscriptionPlanResponse = await this.fetch()
+    const isValidResponse = message === NOTIFIER_RESPONSE_MESSAGES.OK
+
+    if (isValidResponse && !length) {
+      this.errorReporter({
+        error: new Error('No available subscription plan'),
+        text: 'Atleast one subscription plan must be available for registration',
```
Comment (itofarina) [inline src/api/rif-notifier-service/subscriptionPlans.ts]:
```suggestion
        text: 'At least one subscription plan must be available for registration',
```

**Review by itofarina — APPROVED** [review]
LGTM - Please fix the typo.

**Pull request merged**